contracts/0.8.25/vaults/OperatorGrid.sol

```
@@ -0,0 +1,595 @@
+        return _getStorage().tiers[_tierId];
+    }
+
+    /// @notice Alters a tier
+    /// @dev We do not enforce to update old vaults with the new tier params, only new ones.
+    /// @param _tierId id of the tier
+    /// @param _tierParams new tier params
+    function alterTier(uint256 _tierId, TierParams calldata _tierParams) external onlyRole(REGISTRY_ROLE) {
+        ERC7201Storage storage $ = _getStorage();
+        Tier storage tier_ = $.tiers[_tierId];
+        if (_tierId != DEFAULT_TIER_ID && $.tiers[_tierId].operator == address(0)) revert TierNotExists();
+
+        tier_.shareLimit = uint96(_tierParams.shareLimit);
+        tier_.reserveRatioBP = uint16(_tierParams.reserveRatioBP);
+        tier_.rebalanceThresholdBP = uint16(_tierParams.rebalanceThresholdBP);
+        tier_.treasuryFeeBP = uint16(_tierParams.treasuryFeeBP);
+
+        emit TierUpdated(_tierId, tier_.shareLimit, tier_.reserveRatioBP, tier_.rebalanceThresholdBP, tier_.treasuryFeeBP);
+    }
+
+    /// @notice Registers a new vault
+    /// @param vault address of the vault
+    function registerVault(address vault) external {
+        if (vault == address(0)) revert ZeroArgument("_vault");
+
+        ERC7201Storage storage $ = _getStorage();
+        VaultTier storage vaultTier = $.vaultTier[vault];
+        if (vaultTier.currentTierId > 0) revert VaultExists();
+
+        vaultTier.currentTierId = DEFAULT_TIER_ID;
+
+        emit VaultAdded(vault);
+    }
+
+    /// @notice Request to change tier
+    /// @param _vault address of the vault
+    /// @param _tierId id of the tier
+    function requestTierChange(address _vault, uint256 _tierId) external {
+        if (_vault == address(0)) revert ZeroArgument("_vault");
+        if (msg.sender != IStakingVault(_vault).owner()) revert NotAuthorized("requestTierChange", msg.sender);
+
+        ERC7201Storage storage $ = _getStorage();
+        if ($.tiers[_tierId].operator == address(0)) revert TierNotExists();
```

(nitpick) for the default tier it would be a bit misleading though not applicable if default tier is zero as suggested